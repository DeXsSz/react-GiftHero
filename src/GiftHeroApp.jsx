import React from 'react'
import AddCategory from './component/AddCategory';
import GifGrid from './component/GifGrid';

const GiftHeroApp = () => {
    const [categories, setCategories] = React.useState(['One Punch']);
    // const handleAdd = (e) => {
    //     e.preventDefault()
    //     setCategories([...categories, 'Hunter'])
    // }
    return (
        <>
            <h2>GiftHeroApp</h2>
            <hr />
            {/* <button onClick={handleAdd}>Agregar</button> */}
            <AddCategory setCategories={setCategories} />
            <ol>
                {
                    categories.map((category) => {
                        return <GifGrid key={category} category={category} />
                    })
                }
            </ol>
        </>
    )
}

export default GiftHeroApp
