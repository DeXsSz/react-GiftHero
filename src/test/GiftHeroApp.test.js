import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import GiftHeroApp from '../GiftHeroApp';

describe('Pruebas con el GiftHeroApp.jsx', () => {
    test('debe hacer match con el snapshot de GiftHeroApp', () => {
        const wrapper = shallow(<GiftHeroApp />);
        expect(wrapper).toMatchSnapshot();
    });
    test('debe mostrar una lista de categoria', () => {
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GiftHeroApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});
