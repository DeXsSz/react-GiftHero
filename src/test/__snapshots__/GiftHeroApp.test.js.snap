// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el GiftHeroApp.jsx debe hacer match con el snapshot de GiftHeroApp 1`] = `
<Fragment>
  <h2>
    GiftHeroApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Pruebas con el GiftHeroApp.jsx debe mostrar una lista de categoria 1`] = `
<Fragment>
  <h2>
    GiftHeroApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
